packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerOptions.kt

```diff
@@ -1,58 +1,119 @@
+      field = value
+    }
+
+  @Field
+  var base64: Boolean = false
 
-      val isBase64 = options[ImagePickerConstants.OPTION_BASE64] as? Boolean ?: false
-      val mediaTypes = MediaTypes.fromString(
-        options[ImagePickerConstants.OPTION_MEDIA_TYPES] as? String ?: "Images"
-      ).ifNull {
-        promise.reject(ImagePickerConstants.ERR_INVALID_OPTION, "Unknown media types: ${options[ImagePickerConstants.OPTION_MEDIA_TYPES]}")
-        return null
+  @Field
+  var exif: Boolean = false
+
+  @Field
+  var mediaTypes: MediaTypes = MediaTypes.IMAGES
+
+  @Field
+  @IntRange(from = 0)
+  var videoMaxDuration: Int = 0
+    set(value) {
+      assertValueGreaterOrEqual(value, 0)
+      field = value
+    }
+
+  @Field
+  var aspect: Pair<Int, Int>? = null
+    set(value) {
+      // TODO(@bbarthec): check if that works
+      if (value != null) {
+        val (first, second) = value
+        assertValueGreaterOrEqual(first, 0)
+        assertValueGreaterOrEqual(second, 0)
       }
-      val isExif = options[ImagePickerConstants.OPTION_EXIF] as? Boolean ?: false
+      field = value
+    }
+
+  /**
+   * @param uri Target [Uri] that the captured media will be saved into.
+   */
+  internal fun toCameraContract(uri: Uri) = CameraContract(
+    uri,
+    mediaTypes.toMimeType(),
+  )
 
-      val videoMaxDuration = options[ImagePickerConstants.OPTION_VIDEO_MAX_DURATION]?.let {
-        if (it is Number && it.toInt() >= 0) {
-          return@let it.toInt()
-        }
+  fun toImageLibraryContract() = ImageLibraryContract(
+    allowsMultipleSelection,
+    mediaTypes.toMimeType(),
+    mediaTypes.toMultipleMimeTypes()
+  )
+}
+
+enum class MediaTypes(val value: String) {
+  IMAGES("Images"),
+  VIDEOS("Videos"),
+  ALL("All");
+
+  internal fun toMimeType(): String {
+    return when (this) {
+      IMAGES -> ImageAllMimeType
+      VIDEOS -> VideoAllMimeType
+      ALL -> AllMimeType
+    }
+  }
+
+  internal fun toMultipleMimeTypes(): Array<String>? {
+    return when (this) {
+      ALL -> arrayOf(
+        ImageAllMimeType,
+        VideoAllMimeType
+      )
+      else -> null
+    }
+  }
 
-        promise.reject(ImagePickerConstants.ERR_INVALID_OPTION, "videoMaxDuration must be a non-negative integer")
-        return null
-      } ?: 0
+  internal fun toFileExtension(): String {
+    return when(this) {
+      VIDEOS -> ".mp4"
+      else -> ".jpeg"
+    }
+  }
 
-      return ImagePickerOptions(quality, isAllowsEditing, forceAspect, isBase64, mediaTypes, isExif, videoMaxDuration)
+  /**
+   * Return [MediaStore]'s intent capture action associated with given media types
+   */
+  internal fun toCameraIntentAction(): String {
+    return when (this) {
+      VIDEOS -> MediaStore.ACTION_VIDEO_CAPTURE
+      else -> MediaStore.ACTION_IMAGE_CAPTURE
     }
   }
+
+  internal companion object {
+    const val ImageAllMimeType = "image/*"
+    const val VideoAllMimeType = "video/*"
+    const val AllMimeType = "*/*"
+  }
```

Comment:
- I think we keep `const val` names uppercase: `IMAGE_ALL_MIME_TYPE`
- Maybe we could create an enum for this? Something like:

```kt
internal enum class PickerMediaTypes(val mimeType: String) {
  IMAGES("image/*"),
  VIDEOS("video/*"),
  ALL("*/*")
}
```